seerep-hdf5/seerep-hdf5-core/src/hdf5-core-general.cpp

```diff
@@ -390,4 +390,48 @@ std::shared_ptr<HighFive::Group> Hdf5CoreGeneral::getHdf5Group(const std::string
   }
 }
 
+void Hdf5CoreGeneral::writeGeodeticLocation(const seerep_core_msgs::geodeticCoordinates geocoords)
+{
+  const std::scoped_lock lock(*m_write_mtx);
+
+  if (!m_file->hasAttribute(GEODETICLOCATION_COORDINATESYSTEM) || !m_file->hasAttribute(GEODETICLOCATION_ELLIPSOID) ||
+      !m_file->hasAttribute(GEODETICLOCATION_ALTITUDE) || !m_file->hasAttribute(GEODETICLOCATION_LATITUDE) ||
+      !m_file->hasAttribute(GEODETICLOCATION_LONGITUDE))
+  {
+    m_file->createAttribute<std::string>(GEODETICLOCATION_COORDINATESYSTEM, geocoords.coordinateSystem);
+    m_file->createAttribute<std::string>(GEODETICLOCATION_ELLIPSOID, geocoords.ellipsoid);
+    m_file->createAttribute<double>(GEODETICLOCATION_ALTITUDE, geocoords.altitude);
+    m_file->createAttribute<double>(GEODETICLOCATION_LATITUDE, geocoords.latitude);
+    m_file->createAttribute<double>(GEODETICLOCATION_LONGITUDE, geocoords.longitude);
+  }
+  else
+  {
+    m_file->getAttribute(GEODETICLOCATION_COORDINATESYSTEM).write(geocoords.coordinateSystem);
+    m_file->getAttribute(GEODETICLOCATION_ELLIPSOID).write(geocoords.ellipsoid);
+    m_file->getAttribute(GEODETICLOCATION_ALTITUDE).write(geocoords.altitude);
+    m_file->getAttribute(GEODETICLOCATION_LATITUDE).write(geocoords.latitude);
+    m_file->getAttribute(GEODETICLOCATION_LONGITUDE).write(geocoords.longitude);
+  }
+  m_file->flush();
+}
+
+std::optional<seerep_core_msgs::geodeticCoordinates> Hdf5CoreGeneral::readGeodeticLocation()
+{
+  const std::scoped_lock lock(*m_write_mtx);
+
+  seerep_core_msgs::geodeticCoordinates geocoords;
+  if (m_file->hasAttribute(GEODETICLOCATION_COORDINATESYSTEM) || m_file->hasAttribute(GEODETICLOCATION_ELLIPSOID) ||
+      m_file->hasAttribute(GEODETICLOCATION_ALTITUDE) || m_file->hasAttribute(GEODETICLOCATION_LATITUDE) ||
+      m_file->hasAttribute(GEODETICLOCATION_LONGITUDE))
```

Comment:
use & instead of |
or rather check each attribute individually

Reply:
if a certain attribute doesn't exist, should i explicitly assign null or it done by default?

Reply:
return `std::nullopt_t` in this function if one of the fields is missing and do not set the sub-msg in the fb / pb message